crates/locutus-core/src/client_events.rs

```diff
@@ -40,11 +41,43 @@ impl Display for ClientId {
 
 type HostIncomingMsg = Result<OpenRequest<'static>, ClientError>;
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct AuthToken(Arc<str>);
+
+impl AuthToken {
+    pub fn as_str(&self) -> &str {
+        &self.0
+    }
+
+    pub fn generate() -> AuthToken {
+        use rand::Rng;
+        let mut rng = rand::thread_rng();
+        let mut token = [0u8; 32];
+        rng.fill(&mut token);
+
+        // encode
+        let hex_chars: &[u8] = b"0123456789abcdef";
```

Comment:
Sicne this the usage of this should be transparent it shouldn't matter too much what encoding it is and for this usecase doing hex was more efficient. But since this shouldn't matter too much anyway you are right consistency is more important.

Changing to base58 which is what we use everywhere (not base32).